### ISSUE TITLE
[Facility Locator] Display cemetery services on Facility Locator

### ISSUE THREAD

**Issue by mmiddaugh** [department-of-veterans-affairs/va.gov-cms#19893]
## Status
**[2025-01-28]** Per Michelle in [Slack](https://dsva.slack.com/archives/C0FQSS30V/p1738089255999059?thread_ts=1738082605.028049&cid=C0FQSS30V), the Statuses and their Definitions are still TBD from NCA.
**[2025-01-27]** On 1/28, Michelle will meet with NCA to finalize label language for statuses and and definitions
**[2025-01-23]** Schema provided, unblocked
**[2024-12-16]** Schema delayed by upstream decisions; should be available 12/24
**[2024-12-11] [Fran]** Michelle reports that Adam at Lighthouse will have the schema to us by Monday, 12/15.

## Description
The Facility Locator allows Veterans to search for national and VA grant-funded cemeteries (State, Territory and Tribal Veterans Cemeteries) but does not include information about the funeral and burial services available. The need to include these details was emphasized during a Subcommittee on Disability Assistance & Memorial Affairs Oversight Hearing on 11/13/2024. ([1:20 mark](https://www.youtube.com/watch?v=hsouKUKgYkM))

The national cemetery data source includes a status field with possible values of "Open", "Call for burial status" (this is used for Closed),  "Cremation Only" or "Under Construction" representing whether or not the cemetery has space for new, subsequent, or cremation only internments. Equivalent data is not yet available in the data source for grant-funded cemeteries. 

Information about burial space is displayed on NCA's [Find a Cemetery page](https://www.cem.va.gov/find-cemetery/) and on cemetery detail pages. 

Once this data is available in the Lighthouse API, we should be able to make the information available in the Facility Locator. MVP may be brief description on Facility Locator search results card and inclusion on the detail page. 

## High-level Requirements
- **Cemetery Search results cards:**
  - Display the following wording as a link to the cemetery detail page:  Find out about available burial space
- **Cemetery Detail page:** 
  - Display one of the following four statuses, along with the standard definition for each. Note that the statuses will not be custom for each cemetery; they will be one standard definition for each status.
    - Open 
    - Call for burial status (this is used for Closed)
    - Cremation Only
    - Under Construction
    - If there is no status available, display on the Detail page under the 'Burial status' header: 'Call for burial status'
- Cemetery offices:
  - At this time we're assuming that offices won't be sent via the API.
- On Cemetery Detail page, change the text "Hours of Operation' to: see this [comment](https://github.com/department-of-veterans-affairs/va.gov-cms/issues/19893#issuecomment-2672380806) below

![Image](https://github.com/user-attachments/assets/4c27ff00-9182-4dd7-ac55-2cd9c1024f7c)

## Additional Information:
  - Lighthouse will provide the schema (they are trying to get it to us before Christmas).
  - LH will provide both the status and definitions in the LH API (this info is already in the data).
  - There are nationals and state. We are trying to harmonize everything regardless of type of cemetery.
  - ~~Lighthouse will not be able to have the office and non-VA granted funded sites excluded from the API before we do the burial status work so we will need to account for locations without a burial status (at least in the short term). Decision TBD on how to accomplish not displaying statuses for them, or?~~ Dawn Pruitt shared that Lisa has confirmed the non-national facilities (admin, non-funded) have been removed successfully ([Slack](https://dsva.slack.com/archives/C02BTJTDFTN/p1736350367907129?thread_ts=1734646861.422239&cid=C02BTJTDFTN)  This means non-VA grant funded veterans cemeteries and any State Veterans Affairs Headquarters that are not an actual cemetery are have been removed.

## Questions & Answers
- **Question**: Will the status and definitions be in the main data? Or services addendum? Or some new endpoint? We will need to update Vets-api if not in the main data. 
  - **Answer**: Michelle believes it's in the main data but will confirm.
- **Question**: What if there is no status?
  - ~~Assume everything will have a status.~~
  - _Update 12.20.2024: Lighthouse will not be able to pull offices and non-grant-funded locations out of the data prior to this work so we should prepare to handle locations with null burial status values_
- What about offices?
  - ~~Offices may get pulled out of the data.~~ See below
  - _Update 12.20.2024: Lighthouse will not be able to pull offices and non-grant-funded locations out of the data prior to this work so we should prepare to handle locations with null burial status values_
-  **Question**: Which of the following would be easier, considering that Lighthouse will not be able to have the office and non-VA granted funded sites excluded from the API before we do the burial status work so we will need to account for locations without a burial status (at least in the short term). (1) to not add the link to view the status if the burial status is empty, OR (2)  to display as burial status is unknown if the burial status is empty. _Note that it may also depend on how LH approaches this in the data._
  - **Answer**: Lighthouse has confirmed that they WILL be able to exclude these types of facilities.
- **Question**: Did Jordan purposely leave off the icons, such as next to Main phone?
  - **Answer**: Yes, we should remove the phone, website, and direction icons. This will help us align with other facility pages, like the [VAMC location pages](https://www.va.gov/south-texas-health-care/locations/) that don't have icons.

### Design

[Facility Locator Cemetery Services page](https://www.figma.com/design/GvNVeG9GAXZaBpny7vNxZP/Facility-Locator?node-id=1355-17020&t=AfVw5EicYQ9npcM7-1)

### Examples
Cemetery pages have a range of descriptions for open and closed, based on new vs subsequent for Veterans vs family members. Definitions will be confirmed with NCA. "Call for status" may be be needed. 

<details>
<summary> Examples </summary>

Fort Missoula Post Cemetery
- Data: status = Closed 
- On the [Find a cemetery page for Montana](https://www.cem.va.gov/find-cemetery/state.asp?STATE=MT) - Burial Space: Closed 
- On the [NCA Fort Missoula Post Cemetery page](https://www.cem.va.gov/cems/nchp/FtMissoula.asp) - [BURIAL SPACE] Fort Missoula Post Cemetery is currently closed to new interments. The only interments that are being performed are subsequent interments for Veterans or eligible family members in an existing gravesite.

Yellowstone National Cemetery
- Data: status = Open
- On the [Find a cemetery page for Montana](https://www.cem.va.gov/find-cemetery/state.asp?STATE=MT) - Burial Space: Open
- On the [NCA Yellowstone National Cemetery page](https://www.cem.va.gov/cems/nchp/Yellowstone.asp) - [BURIAL SPACE] This cemetery has space available to accommodate casketed and cremated remains.

Woodlawn National Cemetery
- Data: status = Cremation only
- On the [Find a cemetery page for New York](https://www.cem.va.gov/find-cemetery/state.asp?STATE=NY) - Burial Space: Open
- On the [NCA Woodlawn National Cemetery page](https://www.cem.va.gov/cems/nchp/Woodlawn.asp) - [BURIAL SPACE] This cemetery has space available for cremated remains in a columbarium only.

</details>

**Dependent on UX ticket (completed):** https://github.com/department-of-veterans-affairs/va.gov-cms/issues/19916

## Designs

- Search card design: https://www.figma.com/design/GvNVeG9GAXZaBpny7vNxZP/Facility-Locator?node-id=1355-20617&t=BaWAyFcGb8TkuYKI-0
- Detail page design: https://www.figma.com/design/GvNVeG9GAXZaBpny7vNxZP/Facility-Locator?node-id=1430-31210&t=BaWAyFcGb8TkuYKI-0

### User story

**AS A** Veteran, family member, or other beneficiary searching for cemeteries on the Facility Locator
**I WANT** to know whether or not the cemeteries in my search results are open to new interments
**SO THAT** I can easily plan a burial. 

### Engineering notes / background
Similar pattern - Search result cards for community emergency and urgent care locations include "Call to confirm services and hours"

### Approved Schema
- Joe Justice has confirmed in [Slack](https://dsva.slack.com/archives/CJ162GDDJ/p1737649317322379?thread_ts=1736945936.948589&cid=CJ162GDDJ) "..yes, based on the dialogue back and forth we are moving forward with."

API id | API Label | Final UI language
--- | --- | ---
BURIALS_OPEN | "Burial space available" | 
BURIALS_CREMATION_ONLY | "Cremation interments only" | 
BURIALS_CALL_FOR_STATUS  | "Call for burial status" (used when facility is Closed for certain services) | 
BURIALS_UNDER_CONSTRUCTION | "Cemetery is under construction" | 

_Note: I [Slacked](https://dsva.slack.com/archives/CJ162GDDJ/p1737754939689489?thread_ts=1736945936.948589&cid=CJ162GDDJ) Joe to have them correct the spelling of interment (shouldn't be internments)._

### Scope / Impact analysis
- The change to placement of the google maps link, and the text change from Main number to Main phone will be made across the board aka VBA and Cemeteries, since is uses the same component. Instead of doing a one-off, it's best to change the component, and the regression VBA and Cemetery facilities during testing

### Roles / assignments
_After functional testing, code review, accessibility review, and design review can happen in parallel._
- [ ] **Functional testing:** @FranECross 
- [ ] **Code review:** 
- [ ] **Design review:** @davidmpickett 
- [ ] **Accessibility review:** @Shamiaalam 
- [ ] **Product acceptance:** @FranECross and @mmiddaugh 

## Acceptance criteria

MVP AC

**Open Questions as of 2024-01-24:**
- Is removal of the icons next to the phone, etc. intentional? Fran to ask Jordan.
  - Yes, removal was intentional - icons should NOT display.
- Do we want to use the statuses provided by Lighthouse, or the Statuses provided by Jordan in the designs? Fran to ask Michelle/Jordan.
  - Per Michelle [Slack](https://dsva.slack.com/archives/C0FQSS30V/p1738089255999059?thread_ts=1738082605.028049&cid=C0FQSS30V), statuses are still TBD from stakeholders
- Did Michelle hear back from stakeholders regarding the plain text definition suggestions offered by CAIA? Fran to ask Michelle
  - Per Michelle [Slack](https://dsva.slack.com/archives/C0FQSS30V/p1738089255999059?thread_ts=1738082605.028049&cid=C0FQSS30V), Definitions are still TBD from stakeholder

- [x] When Lighthouse provides a status for VA cemeteries (regardless of the status), the Facility Locator search results card displays a link just under the phone number: Learn more about burial status.
  - [x] When clicked, the link should open the Facility's detail page
- [x] Displayed on the Cemetery's (Facility Locator) Detail Page: under the heading "Burial space", that cemetery's current burial status will display, with it's standard definition displayed under it:
  - [x] Call for burial status (this is for Closed)
    - [x] This definition displays under the status (per design): [see comment below](https://github.com/department-of-veterans-affairs/va.gov-cms/issues/19893#issuecomment-2672380806)
  - [x] Cremation Only
    - [x] This definition displays under the status (per design):  [see comment below](https://github.com/department-of-veterans-affairs/va.gov-cms/issues/19893#issuecomment-2672380806)
  - [x] Open
    -  [x] This definition displays under the status (per design):  [see comment below](https://github.com/department-of-veterans-affairs/va.gov-cms/issues/19893#issuecomment-2672380806)
  - [x] Under Construction
    -  [x] This definition displays under the status (per design):  [see comment below](https://github.com/department-of-veterans-affairs/va.gov-cms/issues/19893#issuecomment-2672380806)
  - [x] If there is no status from Lighthouse, display 'Call for availability'. No definition will display under this wording.
  - [x] the component with the 'Get directions', 'Main phone', and (i think) 'Visit cemetery...' links positions changed, which will affect Facility Locator detail pages, regardless of facility type, and they have been tested/confirmed too.
     -  [x] Links/text on the Detail pages have been reordered and some text changed/new. See Figma for details. 
       - Address
       - Get directions on Google Maps
       - Main phone: 
       - Visit the website 
       - Planning to visit? Please call first as information on this page may change.
         - Note that the icons, such as next to Main phone, have been purposely removed and shouldn't be implemented
- On Cemetery Detail page, the text "Hours of Operation' is now:  [see comment below](https://github.com/department-of-veterans-affairs/va.gov-cms/issues/19893#issuecomment-2672380806)

- [ ] Design review passes - @davidmpickett 
- [ ] A11y review passes - @Shamiaalam 
- [ ] Functional testing using the [#19893 Display cemetery status](https://docs.google.com/spreadsheets/d/1wgIM4oIwAQk-E151z07P3hhZuQJ6mhepv7U5tZjdpEI/edit?gid=752579824#gid=752579824) passes OR failing cases have been reviewed by PM and signed off

- [x] Does this product have an existing regression test plan?
	- [x] Yes, [link](https://docs.google.com/spreadsheets/d/1wgIM4oIwAQk-E151z07P3hhZuQJ6mhepv7U5tZjdpEI/edit?gid=0#gid=0) 
	- [x] Yes (in progress) [link](https://docs.google.com/spreadsheets/d/1wgIM4oIwAQk-E151z07P3hhZuQJ6mhepv7U5tZjdpEI/edit?gid=752579824#gid=752579824)
.

**Comment by eselkin** [comment]
A few things to consider:
1. **Blocked** on data schema from Lighthouse 
2. Need to update vets-api (if it comes in the facility data it's probably ok, but if they put it in one of the sub-queries like `facilities/nca_999/services` we'll need to update the main search on `/facilities_api/v2/va` to do the same query to services that we do on `/facilities_api/v2/va/nca_999` - potentially a 3-5pts and put in an argument to make services lookup conditional --- because we don't always want to make a 2nd query for each query we run on FL)
3. **Blocked** v-w updates on design
4. Update vets-website FL results cards - 3-5pts? Depending on the size of change
5. Update vets-website FL/dynamic pages for nca - 3-5pts? As above

**Comment by FranECross** [comment]
Can potentially combine 4&5 above, with points of 5-8 max. #2 needs to be its own ticket at 3-5 points.

**Comment by FranECross** [comment]
Are these tickets sequential? Other than data from LH and design.

- Design, 4, 5 sequential.
- Can do 3,4,5 before we get data from LH.
- AC: #2 update swagger document/other documentation.

**Comment by FranECross** [comment]
Info will come across from LH in the same API. 

Michelle to confirm if the options/info coming from the detail page is different than what it on the tile.

**Comment by mmiddaugh** [comment]
@FranECross We will likely need a way to include the definition for the status 

I see at least three options 
1- The Facility Locator detail page already links to the relevant NCA cemetery page. The link label could be updated to something like, "See burial status and cemetery details" 
2 - Display the corresponding definition on the Facility Locator detail page (when status is closed, the detail page shows the status and the definition of closed"
3 - Display all definitions on each Facility Locator detail page, regardless of that cemetery's status
 
This might be a good topic for the UX sync?

**Comment by FranECross** [comment]
@FranECross Find out from LH if we'll get info if there is no information available so we can display "Call # for status.

**Comment by FranECross** [comment]
@FranECross Check to see why this isn't on the epic.

**Comment by FranECross** [comment]
@eselkin Do you feel this should be two tickets? Search card and detail pages?

**Comment by FranECross** [comment]
@Agile6MSkinner @eselkin @chriskim2311 Michelle has confirmed that Adam at LH will get us the schema on Monday 12/15, so this can be worked on either in Sprint 19, or if not in that Sprint, definitely in Sprint 20 (goal is release by end of Jan 2025). I've tagged Chris in case Eli isn't able to work on this and it's a possibility for Chris to tackle. cc @mmiddaugh

**Comment by jilladams** [comment]
Noting from 12/16 LH sync: https://trello.com/c/cLCqjjZ7#comment-67609de136baecf0f96e0475, schema delayed, now expected by ~12/24.  (Added to status notes in ticket body)

**Comment by SnowboardTechie** [comment]
Schema has been shared [here](https://dsva.slack.com/archives/C0FQSS30V/p1737649862670969)

**Comment by FranECross** [comment]
@SnowboardTechie or @eselkin Will you please remind me which links on a cemetery's detail page are in the 'one' component that we chatted about, which is used by other facility types? 

Is it the Main phone, website, and get directions? I'm finishing up the regression testing and want to add tests to test other facility types (which I'm assuming will be: VA Health, Urgent care, Emergency care, VA Benefits and Vet Centers? (are the community care facilities also using it?) Thanks! cc @jilladams 

![Image](https://github.com/user-attachments/assets/aa53beb9-f2f7-4fd8-9f96-88ce8b87a76f)

T

**Comment by FranECross** [comment]
Cemetery status [regression ](https://docs.google.com/spreadsheets/d/1wgIM4oIwAQk-E151z07P3hhZuQJ6mhepv7U5tZjdpEI/edit?gid=752579824#gid=752579824)test finished (except for the testing of link orders on other facility types).

**Comment by eselkin** [comment]
The same icons are shown on

NCA
VBA (not modernized)
and
anyone who types in a facility id for any other kind of service (basically a details page can get created for any type of facility that gets data back from Lighthouse -- but we only point people there for the above 2 conditions)

The other facilities' regular, modernized pages all use other components or content-build for their phone number pieces... so you don't have to test modernized pages (Vet Center, VBA modernized, VAMC, etc)

**Comment by eselkin** [comment]
@FranECross ^ forgot to tag you

**Comment by FranECross** [comment]
@SnowboardTechie Quick note for you! I noticed in demo that the 'Get directions on Google Maps' isn't directly under the address (on the detail page), but it should be. 😁

**Comment by SnowboardTechie** [comment]
The [FE](https://github.com/department-of-veterans-affairs/vets-website/pull/34472) and [BE](https://github.com/department-of-veterans-affairs/vets-api/compare/vacms-19893-prepare-for-lh-facility-burial-status?expand=1) has been prepared for this update, but we are awaiting updates to the LH API response and final schema for the statuses before this can go any further.

**Comment by jilladams** [comment]
I added 3 ACs here just to make sure we catch all this before the code merges:

- [ ] Design review passes - @davidmpickett 
- [ ] A11y review passes - @Shamiaalam 
- [ ] Functional testing using the [#19893 Display cemetery status](https://docs.google.com/spreadsheets/d/1wgIM4oIwAQk-E151z07P3hhZuQJ6mhepv7U5tZjdpEI/edit?gid=752579824#gid=752579824) passes OR failing cases have been reviewed by PM and signed off

**Comment by jilladams** [comment]
@mmiddaugh  noted that she has a meeting today with NCA for signoff on plain language

**Comment by jilladams** [comment]
@SnowboardTechie if you had to guess, how many points you think are left to get the code ready enough to test, once we get plain language signoff?

**Comment by SnowboardTechie** [comment]
@jilladams I would say 1 point at most.

**Comment by mmiddaugh** [comment]
<details>
<summary> Update to hours block </summary>

![Image](https://github.com/user-attachments/assets/659fb548-ebdf-4f10-944c-aa28a507fd9b)

</details>

- When facility type = VA cemeteries + hours data is available, the heading of the hours block on the detail page is "Open for visitation" 
- When facility type = VA cemeteries + hours data is not available, no hours heading or block is displayed
- Existing functionality is preserved for all other VA facility types

---

Burial Status = Open
> Available at this cemetery:
> - New burial of caskets in a gravesite.
> - New burial of cremated remains in an in-ground gravesite or placement in an above-ground columbarium niche.
> 
> This cemetery may also have a memorial section or a memorial wall. Memorial areas honor decedents whose remains are not recoverable and are not available for burial. (Examples include remains that are donated to science or cremated remains scattered at sea). Please contact the cemetery for more information.

---

Burial Status = Cremation Only
> Available at this cemetery: New burial of cremated remains in an in-ground gravesite and/or placement in an above-ground columbarium niche.
> 
> - If a Veteran or family member is already buried in an existing casket gravesite here, and there is enough space in that casket gravesite, the cemetery can conduct a new casket burial of another eligible family member (called a “subsequent interment”). If you have a loved one who is already buried in an existing casket gravesite here, please contact the cemetery to find out more about subsequent interment.
> 
> This cemetery may also have a memorial section or a memorial wall. Memorial areas honor decedents whose remains are not recoverable and are not available for burial. (Examples include remains that are donated to science or cremated remains scattered at sea). Please contact the cemetery for more information.

---

Burial Status = Closed
> This cemetery has no new gravesites available.
> - If a Veteran or family member is already buried in an existing casket or cremation gravesite here, and there is enough space in that gravesite, the cemetery can conduct a new burial of an eligible family member (called a “subsequent interment”). If you have a loved one who is already buried in an existing gravesite here, contact the cemetery to find out more about subsequent interment.
> - Sometimes, a casket or cremation gravesite or cremation columbarium niche may become available at a closed cemetery. If you are interested in gravesite availability for a deceased loved one, please contact the cemetery.
> 
> This cemetery may have a memorial section or a memorial wall. Memorial areas honor decedents whose remains are not recoverable and are not available for burial. (Examples include remains that are donated to science or cremated remains scattered at sea). Please contact the cemetery for more information.

---

Burial Status = Under construction

> This cemetery isn’t open yet for burials or visitation while it is under construction.
> [Find a VA national cemetery] 

(link to Facility Locator should be pre-filtered to VA cemetery searches if possible/practical)

**Comment by SnowboardTechie** [comment]
@mmiddaugh @FranECross @jilladams @Shamiaalam this is now ready for testing with this review instance tugboat. http://55ab0251be9a8b9ad4f9c9f8dfc109d3.review.vetsgov-internal/

**Comment by SnowboardTechie** [comment]
The [draft PR](https://github.com/department-of-veterans-affairs/vets-website/pull/34472) to complete this work will be ready for review once QA and A11y testing is complete on the review instance. At that point it may be worth expanding the test coverage a bit to cover some more of the final details.

**Comment by jilladams** [comment]
Launch & test plan updates: https://dsva.slack.com/archives/C03LFSPGV16/p1740506691196149. Tl;dr: shipping to main, will test in Staging on the way and revert from Main if we find any issues that mean it shouldn't roll to prod in tomorrow's deploy.

**Comment by jilladams** [comment]
This is now live in prod: https://dsva.slack.com/archives/C0FQSS30V/p1740599741704719

@FranECross leaving for you to close when you're satisfied that it can be!
